### PR TITLE
tests: enable regression and completion suites for opensuse

### PR DESCRIPTION
**Pull request by sergiocazzolato**
Enabling two test suites for opensuse. 
Also it is included a minor change in the opensuse specs to use %{_libexecdir} instead of /usr/lib.
Upgrade test suite is still disabled because there is not snapd avaiblable to install it with zypper yet.

### CODE REVIEW

File: packaging/opensuse-42.2/snapd.spec [hunk 1]
```diff
@@ -121,7 +121,7 @@ export CXXFLAGS
 # apparmor kernel available in SUSE and Debian. The generated apparmor profiles
 # cannot be loaded into a vanilla kernel. As a temporary measure we just switch
 # it all off.
-%configure --disable-apparmor --libexecdir=/usr/lib/snapd
+%configure --disable-apparmor --libexecdir=%{_libexecdir}/snapd
```
Comment (zyga) [inline packaging/opensuse-42.2/snapd.spec]: This is fine

File: packaging/opensuse-42.2/snapd.spec [hunk 2]
```diff
@@ -164,15 +164,15 @@ make %{?_smp_mflags} -C cmd check
 # Install all the go stuff
 %goinstall
 # TODO: instead of removing it move this to a dedicated golang package
-rm -rf %{buildroot}/usr/lib64/go
-rm -rf %{buildroot}/usr/lib/go
+rm -rf %{buildroot}%{_libexecdir}64/go
```
Comment (zyga) [inline packaging/opensuse-42.2/snapd.spec]: Not sure about this

CC @Conan-Kudo any ideas?

File: packaging/opensuse-42.2/snapd.spec [hunk 3]
```diff
@@ -164,15 +164,15 @@ make %{?_smp_mflags} -C cmd check
...
+rm -rf %{buildroot}%{_libexecdir}/go
 find %{buildroot}
-# Move snapd, snap-exec, snap-seccomp and snap-update-ns into /usr/lib/snapd
-install -m 755 -d %{buildroot}/usr/lib/snapd
-mv %{buildroot}/usr/bin/snapd %{buildroot}/usr/lib/snapd/snapd
-mv %{buildroot}/usr/bin/snap-exec %{buildroot}/usr/lib/snapd/snap-exec
-mv %{buildroot}/usr/bin/snap-update-ns %{buildroot}/usr/lib/snapd/snap-update-ns
-mv %{buildroot}/usr/bin/snap-seccomp %{buildroot}/usr/lib/snapd/snap-seccomp
+# Move snapd, snap-exec, snap-seccomp and snap-update-ns into %{_libexecdir}/snapd
+install -m 755 -d %{buildroot}%{_libexecdir}/snapd
```
Comment (zyga) [inline packaging/opensuse-42.2/snapd.spec]: This is fine

File: packaging/opensuse-42.2/snapd.spec (hunk 2 above)
Comment (zyga) [inline packaging/opensuse-42.2/snapd.spec]: Looking at RPM macros I don't see anything relevant. I _think_ it's weird to do it this way and I'd use /usr/lib64/go for this. Alternatively just use `%{_libdir}`

Comment (Conan-Kudo) [inline packaging/opensuse-42.2/snapd.spec]: This should just be `%_libdir`, though this really should just moved to a dedicated golang subpackage...

**Comment by sergiocazzolato** [comment]
Well the error in the following log is on the refresh-all test on autopkgtests run, which is not related to this change. 
log: https://objectstorage.prodstack4-5.canonical.com/v1/AUTH_77e2ada1e7a84929a74ba3b87153c0ac/autopkgtest-zesty-snappy-dev-image/zesty/amd64/s/snapd/20170814_164330_c433e@/log.gz

**Pull request merged**